src/app/integrations/edit-page/edit-page.component.ts

```diff
@@ -155,27 +162,43 @@ export class IntegrationsEditPage implements OnInit, OnDestroy, AfterViewInit {
         // prompt the user what next?
         this.router.navigate(['save-or-add-step', this.currentFlow.getMiddlePosition()], { relativeTo: this.route });
         break;
+      case 'integration-no-actions':
+        if (child !== 'action-select') {
+          this.router.navigate(['action-select', 0], { relativeTo: this.route });
+        }
+        break;
       case 'integration-no-connections':
         if (child !== 'connection-select') {
           this.router.navigate(['connection-select', 0], { relativeTo: this.route });
         }
         break;
+      case 'integration-action-select':
       case 'integration-connection-select':
         if (!this.currentFlow.integration.steps[this.position]) {
           this._canContinue = false;
         }
         break;
+      case 'integration-selected-action':
+        this.position = event['position'];
+        this.currentFlow.events.emit({
+          kind: 'integration-set-action',
+          position: this.position,
+          action: event['action'],
+        });
+        this._canContinue = true;
+        this.continue();
+        break;
       case 'integration-selected-connection':
         this.position = event['position'];
         this.currentFlow.events.emit({
           kind: 'integration-set-connection',
           position: this.position,
-          connection: event['connection'],
+          action: event['connection'],
```

Comment:
should this be 'connection' instead of action, given this is setting the connection, not the action?